### PR TITLE
chore(docs): setup indexing

### LINKED ISSUE #16: Update indexing documentation to Helix 3
Currently located at https://main--helix-home--adobe.hlx.page/docs/setup-indexing.html, the indexing documentation needs to be updated to Helix 3 and ideally brought to `helix-pages/docs`.

### PR DESCRIPTION
**Pull request by dominique-pfister**
Create setup indexing documentation that is in line with Helix 3.

fixes: https://github.com/adobe/helix-website/issues/16

### CODE REVIEW

File: docs/setup-indexing.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+file in your GitHub repository. After having setup your `fstab.yaml` with a mountpoint that points
+into your SharePoint folder, locate a folder where your query index will be located.
+
+Now enter the target URL of that query index location:
+
+1. Select the folder in the Sharepoint UI
+2. Choose action `Copy Link`, and select `Copy` in the popup window
+3. Paste the link into the `helix-query.yaml`s target property
+4. Remove the query string in that link (the question mark `?` and all that follows)
+5. Append the name of your Excel workbook, e.g. `/query-index.xlsx`
```
Comment (tripodsan) [inline docs/setup-indexing.md]: maybe an example would be good here....or a screenshot

File: docs/setup-indexing.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+When your first page is indexed, the indexer will automatically create a suitable workbook at
+that location.
+
+## Setting up the properties that should be added to the index
+
+In order to decide what properties should be indexed in your HTML pages, look at the existing
+`properties` node in the starter `helix-query.yaml`. Further information on their semantics can be
+found in the [Reference](https://github.com/adobe/helix-index-files#reference) section of the
```
Comment (tripodsan) [inline docs/setup-indexing.md:27]: isn't this repo going away once helix2 is deprecated?

File: docs/setup-indexing.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+page, its endpoint is:
+
+https://admin.hlx3.page/index/`owner`/`repo`/`branch`/`path`
+
+You should get a JSON response, where the `data` node contains the index representation of
+your page.
+
+## Setting up more index configurations
+
+You can have more than index configuration in the same `helix-query.yaml`, where different
```
Comment (tripodsan) [inline docs/setup-indexing.md]:
```suggestion
You can have more than one index configuration in the same `helix-query.yaml`, where different
```

**Review by tripodsan — APPROVED** [review]

File: docs/setup-indexing.md (hunk 2 above)
Comment (dominique-pfister) [reply docs/setup-indexing.md:27]: it is 😞 I'll copy its contents to `helix-pages/docs` as well, then

Comment (rofe) [reply docs/setup-indexing.md:27]: @dominique-pfister Actually, the content should be in Google Drive now. I'll give you access.

Comment (dominique-pfister) [reply docs/setup-indexing.md:27]: @rofe: should *this* file go to `helix-website/docs` as well?

Comment (rofe) [reply docs/setup-indexing.md:27]: @dominique-pfister yes I think going forward, all documentation should be in `helix-website/docs`.

**Comment by dominique-pfister** [comment]
@rofe @tripodsan I added the latest changes for the setup indexing doc - can you check again? I'll move it to `helix-website/docs` after that

File: docs/setup-indexing.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# How to add indexing to your Helix 3 project step by step
+
+This document explains how to setup indexing in your Helix 3 project.
+
+## Setting up an initial query index
+
+If you start with the `helix-pages-starter` project you will have a `helix-query.yaml` configuration
```
Comment (tripodsan) [inline docs/setup-indexing.md:7]:
```suggestion
If you start with the [`helix-project-boilerplate`](https://github.com/adobe/helix-project-boilerplate) project you will have a `helix-query.yaml` configuration
```

File: docs/setup-indexing.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+file in your GitHub repository. After having setup your `fstab.yaml` with a mountpoint that points
+into your SharePoint site or Google Drive, select a folder where your query index should be located,
+and enter the path relative to your mountpoint, e.g.:
```
Comment (tripodsan) [inline docs/setup-indexing.md:10]: _and enter the path..._

where do I enter the path?

File: docs/setup-indexing.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+- `/query-index` for Google Drive
+
+For Sharepoint, the indexer will automatically setup the Excel workbook. For Google Drive, however,
+you have to create the spreadsheet yourself:
+
+1. Create the spreadsheet at the location specified above
+2. In that spreadsheet, create a sheet named `raw_index`
+3. In that sheet, enter the names of the properties in your `helix-query.yaml` in the first row -
+   one property name per cell - and add an entry `path`.
+
```
Comment (tripodsan) [inline docs/setup-indexing.md:22]: add screenshot of how the google sheet should look like

File: docs/setup-indexing.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+## Setting up the properties that should be added to the index
+
+In order to decide what properties should be indexed in your HTML pages, look at the existing
+`properties` node in the starter `helix-query.yaml`. Further information on their semantics can be
+found in the [Reference](https://github.com/adobe/helix-index-files#reference) section of the
```
Comment (tripodsan) [inline docs/setup-indexing.md:27]: helix-index-files is archived.... isn't the information now in helix-indexer?

File: docs/setup-indexing.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+You can have more than one index configuration in the same `helix-query.yaml`, where different
+branches are indexed into different Excel workbooks or Google spreadsheets. Again see the
+[Reference](https://github.com/adobe/helix-index-files#reference) for more details
+
+## Setting up an AWS queue to store indexed records into
+
+This is currently a manual process, that someone from the Helix Project must invoke. Please contact
+some team member and tell them your project's GitHub owner and repository.
+
```
Comment (tripodsan) [inline docs/setup-indexing.md:60]: maybe add the steps an helix-admin needs to do as well here, so we have everything in place.

File: docs/setup-indexing.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+3. In that sheet, enter the names of the properties in your `helix-query.yaml` in the first row -
+   one property name per cell - and add an entry `path`.
+
+## Setting up the properties that should be added to the index
+
+In order to decide what properties should be indexed in your HTML pages, look at the existing
+`properties` node in the starter `helix-query.yaml`. Further information on their semantics can be
+found in the [Reference](https://github.com/adobe/helix-index-files#reference) section of the
+`helix-index-files` project.
+
```
Comment (tripodsan) [inline docs/setup-indexing.md:29]: also add a section explaining how to use `hlx up --print-index` to quickly configure your `helix-query.yaml`

**Comment by rofe** [comment]
@dominique-pfister should we close this PR, given that https://github.com/adobe/helix-website/issues/16 will be implemented in content there? I would like to avoid having the same documentation in multiple places, because they tend to go out of sync very quickly.

**Comment by dominique-pfister** [comment]
> @dominique-pfister should we close this PR, given that [adobe/helix-website#16](https://github.com/adobe/helix-website/issues/16) will be implemented in content there? I would like to avoid having the same documentation in multiple places, because they tend to go out of sync very quickly.

Good point, I'll create another PR there.

File: docs/setup-indexing.md (hunk 9 above)
Comment (tripodsan) [reply docs/setup-indexing.md:29]: you can add a link to the mp4:
https://main--helix-website--adobe.hlx3.page/media_15501e6b07101255cb6ef99077f218666809ec354.mp4

( I already uploaded it to the helix-website mediabus)

**Comment by dominique-pfister** [comment]
This is the new proposal: https://main--helix-website--adobe.hlx3.page/docs/setup-indexing

**Pull request closed without merging**